modules/launcher/src/main/java/org/mule/module/launcher/DeploymentService.java

```diff
@@ -101,12 +123,29 @@
     void deployDomain(URL domainArchiveUrl) throws IOException;
 
     /**
+     * Deploys a domain bundled as a zip from the given URL to the mule container
+     *
+     * @param domainArchiveUrl location of the zip domain file
+     * @param appProperties the properties to override
+     * @throws IOException
+     */
+    void deployDomain(URL domainArchiveUrl, Properties appProperties) throws IOException;
+
+    /**
      * Undeploys and redeploys a domain
      *
      * @param domainName then name of the domain to redeploy
      */
     void redeployDomain(String domainName);
 
+    /**
+     * Undeploys and redeploys a domain
+     *
+     * @param domainName then name of the domain to redeploy
+     * @param appProperties the properties to override
+     */
+    void redeployDomain(String domainName, Properties appProperties);
```

Comment:
if redeploy is not called with appProperties (old method) then still the previously used properties are going to be used?